libfreerdp/core/credssp_auth.c

```diff
@@ -199,36 +199,53 @@ static BOOL credssp_auth_client_init_cred_attributes(rdpCredsspAuth* auth)
 	if (auth->kerberosSettings.kdcUrl)
 	{
 		SECURITY_STATUS status = ERROR_INTERNAL_ERROR;
+		SecPkgCredentials_KdcProxySettingsW* secAttr = NULL;
+		SSIZE_T str_size = 0;
+		ULONG buffer_size = 0;
 
-#ifdef UNICODE
-		SecPkgCredentials_KdcUrlW secAttr = { NULL };
-		secAttr.KdcUrl = ConvertUtf8ToWCharAlloc(auth->kerberosSettings.kdcUrl, NULL);
+		str_size = ConvertUtf8ToWChar(auth->kerberosSettings.kdcUrl, NULL, 0);
+		if (str_size <= 0)
+			return FALSE;
+		str_size++;
+
+		buffer_size = sizeof(SecPkgCredentials_KdcProxySettingsW) + str_size * sizeof(WCHAR);
+		secAttr = calloc(1, buffer_size);
+		if (!secAttr)
+			return FALSE;
+
+		secAttr->Version = KDC_PROXY_SETTINGS_V1;
+		secAttr->ProxyServerLength = str_size * sizeof(WCHAR);
+		secAttr->ProxyServerOffset = sizeof(SecPkgCredentials_KdcProxySettingsW);
 
-		if (!secAttr.KdcUrl)
+		if (ConvertUtf8ToWChar(auth->kerberosSettings.kdcUrl, (WCHAR*)(secAttr + 1), str_size) <= 0)
+		{
+			free(secAttr);
 			return FALSE;
+		}
 
+#ifdef UNICODE
 		if (auth->table->SetCredentialsAttributesW)
```

Comment:
Are you sure that the buffer you supply is valid in both, `UNICODE` and non unicode cases?

Reply:
Yes. It's strange, but that's how microsoft decided to do things